File: opspilot-enterprise/packages/shared-types/src/audit.ts
export type AuditEventType =
  | "approval_created" | "approval_decided" | "approval_expired"
  | "tool_invoked" | "execution_started" | "execution_completed" | "execution_failed"
  | "incident_created" | "incident_resolved"
  | "policy_hit" | "policy_bypassed"
  | "case_archived" | "knowledge_updated"
  | "user_login" | "user_logout" | "config_changed";

export type AuditSeverity = "info" | "warning" | "critical";

export interface AuditLog {
  id: string;
  event_type: AuditEventType;
  severity: AuditSeverity;
  actor: string;
  actor_type: "human" | "agent" | "system";
  action: string;
  resource_type: string;
  resource_id: string;
  resource_name: string;
  outcome: "success" | "failure" | "blocked";
  reason: string | null;
  incident_ref: string | null;
  request_id: string;
  trace_id: string;
  timestamp: string;
  metadata: Record<string, unknown>;
}

File: opspilot-enterprise/packages/shared-schema/src/opspilot_schema/audit.py
from __future__ import annotations
from typing import Optional, Any
from pydantic import BaseModel


class AuditLog(BaseModel):
    id: str
    event_type: str
    severity: str
    actor: str
    actor_type: str
    action: str
    resource_type: str
    resource_id: str
    resource_name: str
    outcome: str
    reason: Optional[str] = None
    incident_ref: Optional[str] = None
    request_id: str
    trace_id: str
    timestamp: str
    metadata: dict[str, Any] = {}
